Mathlib/NumberTheory/NumberField/House.lean

```diff
@@ -0,0 +1,308 @@
+/-
+Copyright (c) 2024 Michail Karatarakis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michail Karatarakis
+-/
+import Mathlib.NumberTheory.SiegelsLemma
+import Mathlib.NumberTheory.NumberField.CanonicalEmbedding.Basic
+import Mathlib.NumberTheory.NumberField.EquivReindex
+
+/-!
+
+# House of an algebraic number
+This file defines the house of an algebraic number `α`, which is
+the largest modulus of its conjugates.
+
+## References
+* [D. Marcus, *Number Fields*][marcus1977number]
+* [Keng, H. L, *Introduction to number theory*][keng1982house]
+
+## Tagshouse
+number field, algebraic number, house
+-/
+
+variable {K : Type*} [Field K] [NumberField K]
+
+namespace NumberField
+
+noncomputable section
+
+open Module.Free FiniteDimensional canonicalEmbedding Matrix Finset
+
+attribute [local instance] Matrix.seminormedAddCommGroup
+
+/-- The house of an algebraic number as the norm of its image by the canonical embedding.-/
+def house (α : K) : ℝ := ‖canonicalEmbedding K α‖
+
+theorem house_eq_sup' (α : K) :
+    house α = univ.sup' univ_nonempty (fun φ : K →+* ℂ ↦ ‖φ α‖₊) := by
+  rw [house, ← coe_nnnorm, nnnorm_eq, ← sup'_eq_sup univ_nonempty]
+
+theorem house_sum_le_sum_house {ι : Type*} (s : Finset ι) (α : ι → K) :
+    house (∑ i in s, α i) ≤ ∑ i in s, house (α i) := by
```

Comment:
```suggestion
    house (∑ i ∈ s, α i) ≤ ∑ i in s, house (α i) := by
```
I never remember if this works or not.

Reply:
this works but does something different, I think?